node-wasm/src/worker.rs

```diff
@@ -0,0 +1,328 @@
+            .collect::<Result<Array, _>>()?)
+    }
+
+    async fn get_last_seen_network_head(&mut self) -> JsValue {
+        to_jsvalue_or_undefined(&self.node.get_network_head_header())
+    }
+
+    async fn get_sampling_metadata(&mut self, height: u64) -> Result<Option<SamplingMetadata>> {
+        Ok(self.node.get_sampling_metadata(height).await?)
+    }
+
+    async fn process_command(&mut self, command: NodeCommand) -> WorkerResponse {
+        match command {
+            NodeCommand::IsRunning => WorkerResponse::IsRunning(true),
+            NodeCommand::StartNode(_) => {
+                WorkerResponse::NodeStarted(Err(WorkerError::NodeAlreadyRunning))
+            }
+            NodeCommand::GetLocalPeerId => {
+                WorkerResponse::LocalPeerId(self.node.local_peer_id().to_string())
+            }
+            NodeCommand::GetSyncerInfo => WorkerResponse::SyncerInfo(self.get_syncer_info().await),
+            NodeCommand::GetPeerTrackerInfo => {
+                let peer_tracker_info = self.node.peer_tracker_info();
+                WorkerResponse::PeerTrackerInfo(peer_tracker_info)
+            }
+            NodeCommand::GetNetworkInfo => {
+                WorkerResponse::NetworkInfo(self.get_network_info().await)
+            }
+            NodeCommand::GetConnectedPeers => {
+                WorkerResponse::ConnectedPeers(self.get_connected_peers().await)
+            }
+            NodeCommand::SetPeerTrust {
+                peer_id,
+                is_trusted,
+            } => WorkerResponse::SetPeerTrust(self.set_peer_trust(peer_id, is_trusted).await),
+            NodeCommand::WaitConnected { trusted } => {
+                WorkerResponse::Connected(self.wait_connected(trusted).await)
+            }
+            NodeCommand::GetListeners => WorkerResponse::Listeners(self.get_listeners().await),
+            NodeCommand::RequestHeader(query) => {
+                WorkerResponse::Header(self.request_header(query).await.into())
+            }
+            NodeCommand::GetHeader(query) => {
+                WorkerResponse::Header(self.get_header(query).await.into())
+            }
+            NodeCommand::GetVerifiedHeaders { from, amount } => {
+                WorkerResponse::Headers(self.get_verified_headers(from, amount).await.into())
+            }
+            NodeCommand::GetHeadersRange {
+                start_height,
+                end_height,
+            } => WorkerResponse::Headers(
+                self.get_headers_range(start_height, end_height)
+                    .await
+                    .into(),
+            ),
+            NodeCommand::LastSeenNetworkHead => {
+                WorkerResponse::LastSeenNetworkHead(self.get_last_seen_network_head().await)
+            }
+            NodeCommand::GetSamplingMetadata { height } => {
+                WorkerResponse::SamplingMetadata(self.get_sampling_metadata(height).await)
+            }
+            NodeCommand::CloseWorker => {
```

Comment:
not sure where to put this, but re-opening lumina will result in worker never connecting to the bootstrap peer again. It only works the first time. The only thing that worked for me is reopening chromium

Reply:
I can see it trying to connect but doesn't get any response
![image](https://github.com/eigerco/lumina/assets/34972409/315ce389-c270-495f-9e1c-e84142c2a5ec)

Reply:
or rather we never manage to send the request? :thinking:

Reply:
I've not been able to drill down to the cause of this issue, but using dedicated workers [just added](https://github.com/eigerco/lumina/pull/265#discussion_r1603211525), seems to be a workaround to this issue.
I've added code to default to dedicated worker for chrome and with it I can't seem to trigger the issue.